nvaccess/nvda
# Add new commands for synth ring

### Link to issue number:

Fixes #13768
It's very related to this issue, since it adds these same suggested commands in it for easier navigation in certain cases.

### Summary of the issue:

More efficient navigation in the synth settings ring by adding new commands for it.

### Description of user facing changes

Added scripts:
* Set the current value to the first one as unassigned gesture.
* CTRL+shift+NVDA+page up (laptop) and CTRL+NVDA+page up (desktop) to jump forward the current value by 4x.
* CTRL+shift+NVDA+page down (laptop) and CTRL+NVDA+page down (desktop) to jump backward the current value by 4x.
* Set the current value to the last one as unassigned gesture.

### Description of development approach

In the synth ring, the first and last functions, as increase or decrease by 4x defs as well, (which handles self.min and self.max) were added to both the numeric ring and the global ring, in their respective classes. I consider this approach to be better, because by jumping in this way, focusing on the minimum and maximum values, it can respect the settings of other synth drivers and thus not cause problems with them.

### Testing strategy:

1. Set the synth ring to a specific setting. EX: speed.
2. press control+shift+NVDA+page up or page down (laptop) or control+NVDA+page up or page down (desktop) to jump forward or backward through the currently selected setting. It should increase the value from 20 by 20. Also, assigning a gesture to jump to the first and last value, for example in the rate option, it should set the rate to 100 or 0.

### Known issues with pull request:

None

### Code Review Checklist:



  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Unit tests
  - System (end to end) tests
  - Manual testing
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
